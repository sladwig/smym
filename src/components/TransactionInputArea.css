.transaction-input-area{
    top: 0;
    height: 104px;
    max-height: 104px;
    flex: 0 0 auto;
    width: 100%;
    padding-left:62px;
    padding-right:62px;
    display: flex;
    flex-direction: row;
    justify-content:space-between;
    align-items: center;
    background-color: #ffffff;
}


.top-wrapper {
    position: fixed;
    top: 0;
    display: flex;
    flex-direction: column;
    width: 100%;
    justify-content: flex-start;
    align-items: flex-start;
    box-shadow: 0 1px 0 0 var(--pale-grey);
    transition: all 200ms ease-out;
    z-index: 20;
    background-color: #ffffff;
}
.top-wrapper.active {
    box-shadow: 0 4px 52px 0 rgba(159, 159, 159, 0.14);
}
.top-wrapper.active.has-suggestion {
    box-shadow: 0 36px 60px 0 rgba(74, 80, 87, 0.13);
    border-bottom-left-radius: 42px;
    border-bottom-right-radius: 42px;
}
