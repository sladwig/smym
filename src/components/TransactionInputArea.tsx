import React, { useState, useEffect, useCallback } from 'react';
import './TransactionInputArea.css';
import { Callback } from 'keyboardjs';
import { useHasInLocalStorage } from '../hooks/useLocalStorage';
import { tokenInputFocus } from './TokenDisplay';
import { inputStore, TokenInput } from './TokenInput';
import { useSuggestionStore, SuggestionBox } from './SuggestionBox';
import { MovingEye } from './MovingEye';
import classnames from 'classnames';
import { SearchIcon, SearchSvg } from './Icon';
import { SubmitCancelButton } from './Button';

interface InputAreaProps {
    create: Callback;
}
export const TransactionInputArea = ({ create }: InputAreaProps) => {
    const hasApiToken = useHasInLocalStorage('apiToken');
    const [active, setActive] = useState(false);
    useEffect(() => {
        if (active) tokenInputFocus();
    }, [active]);

    const activate = useCallback(() => {
        if (hasApiToken) setActive(true);
    }, [setActive, hasApiToken]);
    const deactivate = useCallback(() => setActive(false), [setActive]);
    const activateOrDeactivate = useCallback(
        (e: KeyboardEvent) => {
            activate();
            if (e.key.toLowerCase() !== 'escape') return;
            deactivate();
            inputStore.getState().reset();
        },
        [activate, deactivate],
    );

    useEffect(() => {
        document.addEventListener('keydown', activateOrDeactivate);
        return () => {
            document.removeEventListener('keydown', activateOrDeactivate);
        };
    }, [activateOrDeactivate]);

    const mode = useSuggestionStore(state => state.mode);
    const length = useSuggestionStore(state => state.length);
    const suggestion = mode !== 'none' && 0 < length;

    const height = suggestion ? 104 + 90 * (length ? length + 1 : 0) : 104;
    return (
        <div
            className={classnames('top-wrapper', { active, 'has-suggestion': suggestion })}
            style={{ height }}
        >
            <div className={classnames('transaction-input-area', 'heading-1')} onClick={activate}>
                {!active && (
                    <>
                        <MovingEye />
                        <div className="show-container">
                            <div className="show-me-your-money"></div>
                        </div>
                        {hasApiToken && <SearchIcon onClick={activate} />}
                    </>
                )}
                {active && (
                    <>
                        <SearchSvg style={{ marginRight: 29 }} />
                        <TokenInput />
                        <SubmitCancelButton
                            onSubmit={e => {
                                create();
                                deactivate();
                                inputStore.getState().reset();
                                e.stopPropagation();
                            }}
                            onCancel={e => {
                                deactivate();
                                inputStore.getState().reset();
                                e.stopPropagation();
                            }}
                        />
                    </>
                )}
            </div>
            <SuggestionBox />
        </div>
    );
};
