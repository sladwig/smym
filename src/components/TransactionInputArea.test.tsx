import { render } from '@testing-library/react';
import React from 'react';
import { TransactionInputArea } from './TransactionInputArea';

describe('TransactionInputArea', () => {
    test('renders', () => {
        const { container } = render(<TransactionInputArea />);
        expect(container).toMatchSnapshot();
    });
});
