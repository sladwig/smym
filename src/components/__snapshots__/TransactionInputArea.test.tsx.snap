// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionInputArea renders 1`] = `
<div>
  <div
    class="top-wrapper"
    style="height: 104px;"
  >
    <div
      class="transaction-input-area heading-1"
    >
      <div
        class="logo"
      >
        <svg
          height="38"
          viewBox="0 0 38 38"
          width="38"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <circle
              cx="19"
              cy="19"
              fill="#262F5E"
              r="19"
            />
            <path
              d="M19,26 C23.418278,26 27.418278,23.8333333 31,19.5 C27.418278,15.1666667 23.418278,13 19,13 C14.581722,13 10.581722,15.1666667 7,19.5 C10.581722,23.8333333 14.581722,26 19,26 Z"
              fill="#FFF"
            />
            <circle
              cx="20.5"
              cy="20.211498973305954"
              fill="#262F5E"
              r="4.5"
            />
          </g>
        </svg>
      </div>
      <div
        class="show-container"
      >
        <div
          class="show-me-your-money"
        />
      </div>
    </div>
  </div>
</div>
`;
